dotnetv3/AutoScaling/Actions/AutoScalingWrapper.cs

```diff
@@ -0,0 +1,353 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+// snippet-start:[AutoScaling.dotnetv3.AutoScalingActions.AutoScalingWrapper]
+using LaunchTemplateSpecification = Amazon.AutoScaling.Model.LaunchTemplateSpecification;
+
+namespace AutoScalingActions;
+
+/// <summary>
+/// A class that includes methods to perform Amazon EC2 Auto Scaling
+/// actions.
+/// </summary>
+public class AutoScalingWrapper
+{
+    private readonly IAmazonAutoScaling _amazonAutoScaling;
+
+    /// <summary>
+    /// Constructor for the AutoScalingWrapper class.
+    /// </summary>
+    /// <param name="amazonAutoScaling">The injected Amazon EC2 Auto Scaling client.</param>
+    public AutoScalingWrapper(IAmazonAutoScaling amazonAutoScaling)
+    {
+        _amazonAutoScaling = amazonAutoScaling;
+    }
+
+    // snippet-start:[AutoScaling.dotnetv3.AutoScalingActions.CreateAutoScalingGroup]
+
+    /// <summary>
+    /// Create a new Amazon EC2 Auto Scaling group.
+    /// </summary>
+    /// <param name="groupName">The name to use for the new Auto Scaling
+    /// group.</param>
+    /// <param name="launchTemplateName">The name of the Amazon EC2 Auto Scaling
+    /// launch template to use to create instances in the group.</param>
+    /// <param name="serviceLinkedRoleARN">The AWS Identity and Access
+    /// Management (IAM) service-linked role that provides the permissions
+    /// for the Amazon EC2 Auto Scaling group.</param>
+    /// <returns>A Boolean value indicating the success of the action.</returns>
+    public async Task<bool> CreateAutoScalingGroupAsync(
+        string groupName,
+        string launchTemplateName,
+        string serviceLinkedRoleARN)
+    {
+        var templateSpecification = new LaunchTemplateSpecification
+        {
+            LaunchTemplateName = launchTemplateName,
+        };
+
+        var zoneList = new List<string>
+            {
+                "us-east-2a",
```

Comment:
Is there a way to do this without hard-coding it? Could we load it from config?

Reply:
Yes, it is loaded from the settings.json and instructions for setting the value have been added to the README file.